File: serve-odtlite.py
#!/usr/bin/env python3
import http.server
import socketserver
import os
import sys

PORT = 8000
ODTLITE_PATH = r'E:\ODTLite'

os.chdir(ODTLITE_PATH)

Handler = http.server.SimpleHTTPRequestHandler

print(f"Serving ODTLite from: {ODTLITE_PATH}")
print(f"Open http://localhost:{PORT} in your browser")

with socketserver.TCPServer(("", PORT), Handler) as httpd:
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        print("\nServer stopped.")
        sys.exit(0)

File: serve-odtlite.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = 8000;
const ODTLITE_PATH = 'E:\\ODTLite';

const server = http.createServer((req, res) => {
    let filePath = path.join(ODTLITE_PATH, req.url === '/' ? 'index.html' : req.url);
    
    const ext = path.extname(filePath);
    const mimeTypes = {
        '.html': 'text/html',
        '.js': 'application/javascript',
        '.md': 'text/markdown',
        '.css': 'text/css',
        '.json': 'application/json'
    };
    
    fs.readFile(filePath, (err, data) => {
        if (err) {
            res.writeHead(404, {'Content-Type': 'text/html'});
            res.end('<h1>404 Not Found</h1>');
            return;
        }
        
        res.writeHead(200, {'Content-Type': mimeTypes[ext] || 'application/octet-stream'});
        res.end(data);
    });
});

server.listen(PORT, () => {
    console.log(`Serving ODTLite from: ${ODTLITE_PATH}`);
    console.log(`Open http://localhost:${PORT} in your browser`);
});
